numpyro/distributions/discrete.py

```diff
@@ -33,6 +37,9 @@ def _entropy(self, n, p):
 
 
 class _bernoulli_gen(jax_discrete, bernoulli_gen):
+    # TODO: This should depend on initialization
+    arg_constraints = {'p': constraints.real}
```

Comment:
To me, the simplest solution right now is to create a new class, named for example: bernoulli_with_logits. ^^! WDYT?

Reply:
That would be one way to do it, but there are many cases where we can only do validation after initialization, e.g. for binomial, where support is in `[0, n]`. Also the multivariate cases are a mess to handle, so I think it would be preferable prioritizing #82 than invest more time in working around these issues. 

For the new distribution classes, I currently have `probs` and `logits` as separate parametrizations for the same reason, and it is worth thinking if it is better to separate out the implementations.

Reply:
About support, I think we just mimic the way we did in continuous distribution: get arg and verify. I think it is what PyTorch does (except @lazy decorator). For special case like binomial, we can just overwrite that `_support` property I guess.

Reply:
`_support` needs to be defined at the correct place (not in gen, but rather in frozen) because it needs to be treated as a property and not something into which arguments will be passed into. In any case, this isn't an issue with the classes in `contrib`, so I'm just focusing on that for the moment.

Reply:
Ah, I mean to modify the _support method in gen. Anyway, I’ll do that instead. :)